mula/scheduler/schedulers/boefje.py

```diff
@@ -242,6 +242,16 @@ def push_tasks_for_new_boefjes(self) -> None:
         )
 
         for boefje in new_boefjes:
+            if not boefje.consumes:
```

Comment:
Would it make sense to drop any boefje that has no inputs from the list of boefjes before we add it to the local 'boefjes' cache? Thath way we could clean the new_boefjes list before it even gets here.